--- a/net/minecraft/server/dedicated/Settings.java
+++ b/net/minecraft/server/dedicated/Settings.java
@@ -119,7 +_,7 @@
       String string2 = this.getStringRaw(string);
       V object2 = MoreObjects.firstNonNull(string2 != null ? function.apply(string2) : null, object);
       this.properties.put(string, function2.apply(object2));
-      return new Settings.MutableValue<>(string, object2, function2);
+      return new Settings.MutableValue(string, object2, function2);
    }
 
    protected <V> V get(String string, Function<String, V> function, UnaryOperator<V> unaryOperator, Function<V, String> function2, V object) {
