--- a/net/minecraft/util/EncoderCache.java
+++ b/net/minecraft/util/EncoderCache.java
@@ -31,7 +_,7 @@
          public <T> DataResult<T> encode(A object, DynamicOps<T> dynamicOps, T object2) {
             return EncoderCache.this.cache
                .getUnchecked(new EncoderCache.Key<>(codec, object, dynamicOps))
-               .map(objectx -> objectx instanceof Tag tag ? tag.copy() : objectx);
+               .map(objectx -> objectx instanceof Tag tag ? (T) tag.copy() : (T) objectx);
          }
       };
    }
