--- a/net/minecraft/commands/synchronization/SuggestionProviders.java
+++ b/net/minecraft/commands/synchronization/SuggestionProviders.java
@@ -49,7 +_,7 @@
          throw new IllegalArgumentException("A command suggestion provider is already registered with the name " + resourceLocation);
       } else {
          PROVIDERS_BY_NAME.put(resourceLocation, suggestionProvider);
-         return new SuggestionProviders.Wrapper(resourceLocation, suggestionProvider);
+         return (SuggestionProvider<S>) new Wrapper(resourceLocation, suggestionProvider);
       }
    }
 
