--- a/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -44,7 +_,7 @@
 
    private static <T> Function<Typed<?>, Typed<?>> fixChoiceCap(String string, FixProjectileStoredItem.SubFixer<?> subFixer, Type<?> type, Type<T> type2) {
       OpticFinder<?> opticFinder = DSL.namedChoice(string, type);
-      return typed -> typed.updateTyped(opticFinder, type2, typedx -> subFixer.fix(typedx, type2));
+      return typed -> typed.updateTyped(opticFinder, type2, typedx -> subFixer.fix(typedx, (Type)type2));
    }
 
    private static <T> Typed<T> fixArrow(Typed<?> typed, Type<T> type) {
